web/twa-vis-platform/code/src/ui/map/mapbox/mapbox-container.tsx

```diff
@@ -29,67 +29,84 @@ interface MapProperties {
  * @returns React component for display.
  */
 export default function MapboxMapComponent(props: MapProperties) {
-  const mapRef = useRef();
-  const mapContainerRef = useRef();
+    const mapContainerRef = useRef();
 
-  // Run when component loaded
-  useEffect(() => {
-    initialiseMap();
+    // Run when component loaded
+    useEffect(() => {
+        initialiseMap();
 
-    return () => {
-      if (props.currentMap) {
-        props.currentMap.remove(); // Remove the map instance
-        props.setMap(null); // Reset the map ref
-      }
-    };
-  }, []);
+        return () => {
+            if (props.currentMap) {
+                props.currentMap.remove(); // Remove the map instance
+                props.setMap(null); // Reset the map ref
+            }
+        };
+    }, []);
 
-  // Initialise the map object
-  const initialiseMap = async () => {
-    props.currentMap?.remove();
+    // Initialise the map object
+    const initialiseMap = async () => {
+        props.currentMap?.remove();
 
-    const defaultImagery: ImageryOption = props.imageryOption ?? {
-      "name": "Standard (Night)",
-      "url": "mapbox://styles/mapbox/standard",
-      "time": "dusk"
-    };
+        const defaultImagery: ImageryOption = props.imageryOption ?? {
+            "name": "Standard (Night)",
+            "url": "mapbox://styles/mapbox/standard",
+            "time": "dusk"
+        };
 
-    mapboxgl.accessToken = process.env.MAPBOX_API_KEY;
+        mapboxgl.accessToken = process.env.MAPBOX_API_KEY;
 
-    const map: Map = new mapboxgl.Map({
-      container: mapContainerRef.current,
-      style: defaultImagery.url,
-      center: props.defaultPosition.center,
-      zoom: props.defaultPosition.zoom,
-      bearing: props.defaultPosition.bearing,
-      pitch: props.defaultPosition.pitch,
-    });
+        const map: Map = new mapboxgl.Map({
+            container: mapContainerRef.current,
+            style: defaultImagery.url,
+            center: props.defaultPosition.center,
+            zoom: props.defaultPosition.zoom,
+            bearing: props.defaultPosition.bearing,
+            pitch: props.defaultPosition.pitch,
+            transformRequest: (url) => {
+                if (process.env.REACT_APP_USE_GEOSERVER_PROXY === 'true') {
+                    try {
+                        const urlObject = new URL(url);
+                        const params = new URLSearchParams(urlObject.search);
+                        if (params.get('request') === 'GetMap') {
+                            // not sure if this will work across all conditions
+                            const serverUrl = `${window.location.protocol}//${window.location.host}`;
+                            const proxyUrl = `${serverUrl}/geoserver-proxy?url=${encodeURIComponent(url)}`;
```

Comment:
this needs to change as we do not want to restrict viz and geoserver url to the same host as the viz url

Reply:
Will add an additional environment variable for the server url

Reply:
agree with this

Reply:
addressed by https://github.com/cambridge-cares/TheWorldAvatar/pull/1384/commits/1250573f4f2596ebae7ea3d0953f83696bb369fe